### DISCUSSION TITLE
Export both server components and client components in one barrel file

### DISCUSSION THREAD

**Discussion by mrcaidev** [vercel/next.js#64111] [Help]
### Summary

Is it wise to export both server components and client components in one barrel file?

```ts
// index.ts
export { ServerComponent } from "./server-component";
export { ClientComponent } from "./client-component";
```

I wonder if Next.js could correctly tree-shake them, so that server components wouldn't get shipped in client bundle, or vice versa.

**Comment by akaGooddevil** [comment]
👀👀👀
